app/src/main/java/com/vultisig/wallet/presenter/import_file/ImportFileScreen.kt

```diff
@@ -30,28 +35,59 @@ import androidx.compose.ui.geometry.CornerRadius
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.graphics.PathEffect
 import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.core.graphics.toColorInt
+import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.rememberNavController
 import com.vultisig.wallet.R
+import com.vultisig.wallet.common.fileName
+import com.vultisig.wallet.common.fileContent
+import com.vultisig.wallet.presenter.import_file.ImportFileEvent.*
+import com.vultisig.wallet.presenter.import_file.ImportFileUiEvent.CopyFileToAppDir
+import com.vultisig.wallet.presenter.import_file.ImportFileUiEvent.FetchFileName
 import com.vultisig.wallet.ui.components.TopBar
-import com.vultisig.wallet.ui.navigation.Screen
-import com.vultisig.wallet.ui.theme.appColor
+import com.vultisig.wallet.ui.theme.Theme
 import com.vultisig.wallet.ui.theme.dimens
-import com.vultisig.wallet.ui.theme.menloFamily
-import com.vultisig.wallet.ui.theme.montserratFamily
 
 @Composable
-fun ImportFile(navController: NavHostController, hasFile: Boolean) {
-    val textColor = MaterialTheme.appColor.neutral0
+fun ImportFileScreen(navController: NavHostController) {
+    val context = LocalContext.current
+    val viewmodel = hiltViewModel<ImportFileViewModel>()
+    val uiModel by viewmodel.uiModel.collectAsState()
+    val appColor = Theme.colors
+    val menloFamily = Theme.menlo
+    val montserratFamily = Theme.montserrat
+
+    val launcher =
+        rememberLauncherForActivityResult(ActivityResultContracts.GetContent()) { uri: Uri? ->
+            viewmodel.onEvent(FileSelected(uri))
+        }
+
+    LaunchedEffect(key1 = Unit) {
+        viewmodel.channel.collect { uiEvent ->
+            when (uiEvent) {
+                is CopyFileToAppDir -> {
+                    val fileContent = uiEvent.uri.fileContent(context)
+                    viewmodel.onEvent(FileContentFetched(fileContent))
+                }
+
+                is FetchFileName -> {
+                    val fileName = uiEvent.uri.fileName(context)
+                    viewmodel.onEvent(FileNameFetched(fileName))
```

Comment:
I think you can inject ContentResolver from the application's context into ViewModel to avoid this back-and-forth

Reply:
While injecting the ContentResolver from the application's context into the ViewModel may seem like a solution to avoid back-and-forth communication, it goes against the recommended architecture guidelines for Android development. According to the official Android documentation on UI architecture (https://developer.android.com/topic/architecture/ui-layer#logic-types), UI logic, especially when it involves UI types like Context, should reside in the UI layer, not in the ViewModel. This approach ensures better separation of concerns, improves testability, and makes the codebase easier to maintain in the long run.

Reply:
Well, first of all, it's not UI logic. 
You work with files here; it hardly has anything to do with UI.

Second, ViewModel is in the UI layer. That's how it is on the diagrams, just a bit above where your link leads.
<img width="516" alt="image" src="https://github.com/vultisig/vultisig-android/assets/14962060/8c15ea7c-c807-4bfa-94aa-fa5230a9a846">

Using ApplicationContext is all well and good in the layers under ViewModel. 
However, if you want to make it a really clean architecture, move injecting context to the Repository in the data layer. 

The current code just induces recompositions for working with files and complicates the UI logic.